--- origsrc/cairo-dock-2.1.3-3/src/cairo-dock-modules.c	2010-02-13 04:55:15.000000000 -0600
+++ src/cairo-dock-2.1.3-3/src/cairo-dock-modules.c	2010-02-17 02:02:15.854041400 -0600
@@ -157,7 +157,11 @@ static gchar *cairo_dock_extract_default
 		ptr = cSoFilePath;
 	else
 		ptr ++;
+#ifdef __CYGWIN__
+	if (strncmp (ptr, "cyg", 3) == 0)
+#else
 	if (strncmp (ptr, "lib", 3) == 0)
+#endif
 		ptr += 3;
 
 	if (strncmp (ptr, "cd-", 3) == 0)
@@ -167,7 +171,7 @@ static gchar *cairo_dock_extract_default
 
 	gchar *cModuleName = g_strdup (ptr);
 
-	ptr = g_strrstr (cModuleName, ".so");
+	ptr = g_strrstr (cModuleName, "."G_MODULE_SUFFIX);
 	if (ptr != NULL)
 		*ptr = '\0';
 
@@ -364,7 +368,7 @@ void cairo_dock_preload_module_from_dire
 		if (cFileName == NULL)
 			break ;
 		
-		if (g_str_has_suffix (cFileName, ".so"))
+		if (g_str_has_suffix (cFileName, "."G_MODULE_SUFFIX))
 		{
 			g_string_printf (sFilePath, "%s/%s", cModuleDirPath, cFileName);
 			pModule = cairo_dock_load_module (sFilePath->str, &tmp_erreur);
