--- origsrc/cairo-dock-2.2.0-2/src/gldit/cairo-dock-modules.c	2010-09-18 06:53:14.000000000 -0500
+++ src/cairo-dock-2.2.0-2/src/gldit/cairo-dock-modules.c	2010-09-19 19:32:37.559943600 -0500
@@ -180,7 +180,11 @@ static gchar *cairo_dock_extract_default
 		ptr = cSoFilePath;
 	else
 		ptr ++;
+#ifdef __CYGWIN__
+	if (strncmp (ptr, "cyg", 3) == 0)
+#else
 	if (strncmp (ptr, "lib", 3) == 0)
+#endif
 		ptr += 3;
 
 	if (strncmp (ptr, "cd-", 3) == 0)
@@ -190,7 +194,7 @@ static gchar *cairo_dock_extract_default
 
 	gchar *cModuleName = g_strdup (ptr);
 
-	ptr = g_strrstr (cModuleName, ".so");
+	ptr = g_strrstr (cModuleName, "."G_MODULE_SUFFIX);
 	if (ptr != NULL)
 		*ptr = '\0';
 
@@ -391,7 +395,7 @@ void cairo_dock_load_modules_in_director
 		if (cFileName == NULL)
 			break ;
 		
-		if (g_str_has_suffix (cFileName, ".so"))
+		if (g_str_has_suffix (cFileName, "."G_MODULE_SUFFIX))
 		{
 			g_string_printf (sFilePath, "%s/%s", cModuleDirPath, cFileName);
 			pModule = cairo_dock_load_module (sFilePath->str, &tmp_erreur);
